UnitTests/TestHelper.cs

```diff
@@ -6,4 +9,10 @@ public static class TestHelper
 {
     public static IEnumerable<Observation> Observation =>
         ((InMemoryObservationPublisher)Scientist.ObservationPublisher).Observations;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static IQbservable<Observation> ObservationsGeneratedInThisMethod([CallerMemberName] string callingMethodName = "")
```

Comment:
Helper function to find  observations created in the same method which this function was called.